Modify rule S1301: LaYC - switch replacable by if (#3111)
SonarSource/rspec: rules/S1301/cfamily/rule.adoc

== Why is this an issue?

`switch` statements are useful when there are many different cases depending on the value of the same expression. For just one or two cases, however, the code will be more readable with `if` statements.

In particular, `if` statements are more suitable when the condition of the `switch` is boolean.

> This rule detects statements that could be simplified with a simple `if / else`. That is when there is no more than one `case`, not counting those that fall through to a `default`.

Reviewer: ```suggestion
This rule detects statements that could be simplified with a simple `if / else`. That is when there is at most one `case`, not counting those that fall through to a `default`.
```

The following code:

[source,cpp]
----
switch (variable) {
  case 0:
    doSomething();
    break;
  case 1:
  case 2:
  default:
    doSomethingElse();
    break;
}
----

Would be more readable that way:

[source,cpp]
----
if (variable == 0) {
  doSomething();
} else {
  doSomethingElse();
}
----

> While the following codes don't trigger the rule, because using `if` would not improve their readability:

Reviewer: "Code" is a mass noun in this context. Snippets may be better. There is also an extra comma.

```suggestion
While the following snippets don't trigger the rule because using `if` would not improve their readability:
```
